cdap-docs/developers-manual/source/building-blocks/datasets/index.rst

```diff
@@ -125,9 +125,19 @@ There are two ways to use a dataset in a program:
   by the program.
 
   However, contrary to static datasets, dynamic datasets allow the release of the resources held by their Java classes
-  after you are done using them. You can do that by calling the ``discardDataset()`` method of the program context:
+  after you are finished using them. You can do that by calling the ``discardDataset()`` method of the program context:
   it marks the dataset to be closed and removed from all transactions as soon as the current transaction completes,
-  thereby releasing its resources.
+  thereby releasing its resources. Note that this does not release the dataset immediately: it only marks it for removal
+  upon completion of the current transaction. This is because the dataset must still participate in the committing (or
+  rollback) of the transaction. Also, after calling ``discardDataset()``, you should not perform any further operations
+  on the dataset.
+
+  Because discarding a dataset removes it from the program's cache, it is also a useful method of refreshing a dataset
+  after its properties have been updated: the next time you call ``getDataset()`` (in a new transaction), it is
+  guaranteed that the dataset is instantiated freshly, thereby picking up any changes in the dataset properties that
+  were made after the program started. Be aware that if you call ``getDataset()`` again for the same dataset and
+  arguments before the transaction is complete, then that reactivates the dataset, reversing the effect of discarding
+  it. It is therefore a good practice to discard a dataset at the end of a transaction.
```

Comment:
It is scenario 1. It is exactly the scenario that addresses the Jira. discard() immediately followed by get(). I will try to rephrase so that you understand better.